src/common/dao/register.go

```diff
@@ -50,8 +50,8 @@ func Register(user models.User) (int64, error) {
 // UserExists returns whether a user exists according username or Email.
 func UserExists(user models.User, target string) (bool, error) {
 
-	if user.Username == "" && user.Email == "" {
-		return false, errors.New("user name and email are blank")
+	if user.Username == "" && user.Email == "" && user.Realname == "" {
```

Comment:
can we add a comment here explaining that we use `Readname` for the user ID as it exists in kubernetes-auth